### DISCUSSION TITLE
Marksman with nvim-lspconfig - toml config question

### DISCUSSION THREAD

**Discussion by ghost** [LazyVim/LazyVim#1442] [Q&A] [answered]
Hi all,

The Marksman LSP looks for a `.marksman.toml` file to [tweak a few settings](https://github.com/artempyanykh/marksman/blob/main/docs/configuration.md). I'm pretty sure I can instead configure these settings via nvim-lspconfig (and get rid of the toml file entirely), but I'm struggling to get the right syntax. 

Here's what I currently have. Marksman installs & works fine, but the `toc` feature is still enabled and link completion is still using the default `title-slug` instead of what I have here:

```lua
return {
  "neovim/nvim-lspconfig",
  enabled = true,
  opts = {
    autoformat = true,
    servers = {
      marksman = {
        settings = {
          markdown = {
            code_action = {
              toc = false,
              create_missing_file = false,
            },
            completion = {
              wiki = {
                style = "file-stem",
              },
            },
          },
        },
      },
    },
  },
}
```

for reference, the ol' .toml file looks like this:

```toml
[code_action]
# Enable/disable "Table of Contents" code action
toc.enable = false

[completion]
# The style of wiki links completion.
wiki.style = "file-stem"

```

I'm getting lost with where / how the headings in my toml file get plugged into LazyVim. I'm pretty convinced the issue is somewhere around the `servers = { marksman = { settings = { markdown = {}}}}` area, but I'm not finding the solution. I've pieced my file together by looking at how the [Typescript](https://www.lazyvim.org/extras/lang/typescript#nvim-lspconfig) and [Python](https://www.lazyvim.org/extras/lang/python#nvim-lspconfig) languages are configured, but I expect there's some fundamental piece of LazyVim I'm not grokking.

Any advice would be appreciated!

**Comment by dpetka2001** [comment]
Are you sure that `Marksman` accepts `settings` field? From the original [docs](https://github.com/artempyanykh/marksman/blob/main/docs/configuration.md) it seems that it only supports configuration through `.toml` files.

**Reply by ghost** [reply]
> Are you sure that Marksman accepts settings field?

Not at all. Marksman docs do indeed lead me to believe it only supports configuration via toml, but I was thinking that maybe the `settings` field was somehow interpreted differently through nvim-lspconfig, as I saw it on a few different LSP language config examples.

**Reply by dpetka2001** [reply, marked as answer]
I'm just guessing here, but I believe this depends on what kind of request any server is able to receive. If it can't recognize the `settings` field, then even if `lspconfig` provides such a capability it has no meaning for the specified server. You would have to ask at `Marksman`'s repo if such a capability is available.

**Reply by ghost** [reply]
I think that's a solid guess. I'll go do some digging over there! Thanks for the wisdom, @dpetka2001.

**Reply by dpetka2001** [reply]
This isn't wisdom as I don't know how LSPs work internally. This was just a guess based on observations that might be wrong. Pls take it with a grain of salt and look for more concrete source of information.

**Reply by ghost** [reply]
The wisdom was pointing me towards Marksman's repo 😁👍 I've [opened a discussion](https://github.com/artempyanykh/marksman/discussions/248) there to get further down this rabbit hole.
